src/library/scala/concurrent/Promise.scala

```diff
@@ -17,6 +17,10 @@ import scala.util.{ Try, Success, Failure }
 /** Promise is an object which can be completed with a value or failed
  *  with an exception.
  *
+ *  A promise should always eventually be completed, whether for success or failure, 
+ *  in order to avoid unintended resource retention for any associated Future's 
```

Comment:
I'm late here, but this should be `Futures'` (possessive of multiple `Future`s).

Reply:
Thanks, fixed in https://github.com/scala/scala/commit/b1769e425576f48af711d645a0f9195e465438fd